examples/gitlab/README.md

```diff
@@ -0,0 +1,229 @@
+
+```shell
+npm i uuid
+```
+Install the [cookie-session](https://www.npmjs.com/package/cookie-session) package to allow your application to establish a session (an HTTP-Only cookie) to store the authorizing user id.
+
+```shell
+npm i cookie-session
+```
+
+## Add Routes
+
+There are two critical routes we need to add to our application:
+
+- Authorize route
+- Finished route
+
+Let’s understand the role of each route:
+
+### Authorize route
+
+EveryAuth middleware enables your application to perform an authorization flow for a particular service.
+EveryAuth provides an out-of-the-box shared GitLab OAuth Client so that you can get up and running quickly.
+
+EveryAuth simplifies a lot the authorization flow:
+
+```javascript
+  app.use(
+    '/authorize/:userId',
+    (req, res, next) => {
+      if (!req.params.userId) {
+        return res.redirect('/');
+      }
+      return next();
+    },
+    everyauth.authorize('gitlab’, {
+      // The endpoint of your app where control will be returned afterwards
+      finishedUrl: '/finished',
+      // The user ID of the authenticated user the credentials will be associated with
+      mapToUserId: (req) => req.params.userId,
+   })
+  );
+```
+
+You can define any name you want for the authorization route. In our previous example, it’s called `authorize`, but it’s up to you, and your application needs to use a different name/path. 
+
+### Finished route
+
+After the authorization flow finishes, control is returned to your application by redirecting the user to the configured `finishedUrl` in the `authorize` route.
+The redirection includes query parameters that your application can use to know the [operation status](https://github.com/fusebit/everyauth-express#parameters---2).
+You can use any path for the route. Just ensure it matches what you have configured in the `finishedUrl` property.
+In this route, you can now interact with the Gitlab API by leveraging the EveryAuth service to get a fresh access token.
+ 
+We will get the authorizing GitLab user information and starred repositories using the REST API.
+
+```javascript
+app.get('/finished', handleSession, async (req, res) => {
+  const userCredentials = await everyauth.getIdentity('gitlab', req.session.userId);
+  const gitlabClient = new Gitlab({ oauthToken: userCredentials.accessToken });
+  const user = await gitlabClient.Users.current();
+
+  // Get the last top 10 user starred repositories from GitLab.
+  const projects = await gitlabClient.Projects.all({ maxPages: 1, perPage: 10, starred: true });
+   . . . render the data
+});
+```
+
+Now, we need to display the data. We will use a simple template engine called [pug](https://www.npmjs.com/package/pug), which allows us to quickly render an HTML page with the data returned from GitHub.
+
+Install the dependency and set it as the default view engine:
+
+```shell
+npm i pug
```

Comment:
I want to see less "how to make this example from scratch yourself" content in this README and more "this is what it's doing and the one or two relevant pieces, with a screenshot".

Good illustration, though, that the blog post as whole isn't a good target here.